File: exe4.java
class Animal {
    public void som() {
        System.out.println("O animal faz um som.");
    }
}

class Cachorro extends Animal {
    @Override
    public void som() {
        System.out.println("O cachorro late: Au Au!");
    }
}

class Gato extends Animal {
    @Override
    public void som() {
        System.out.println("O gato mia: Miau!");
    }
}

public class exe4 {
    public static void main(String[] args) {
        Animal cachorro = new Cachorro();
        Animal gato = new Gato();

        cachorro.som();  
        gato.som();  
    }
}

File: exe4.py
class Animal:
    def som(self):
        print("O animal faz um som.")

class Cachorro(Animal):
    def som(self):
        print("O cachorro late: Au Au!")

class Gato(Animal):
    def som(self):
        print("O gato mia: Miau!")

if __name__ == "__main__":
    cachorro = Cachorro()
    gato = Gato()

    cachorro.som()  
    gato.som()    
